core/services/workflows/metering/metering.go

```diff
@@ -195,8 +200,8 @@ func (r *Report) Reserve(ctx context.Context) error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	// to avoid nil pointers, the balance store should be initialized with a zero value
-	r.balance, _ = NewBalanceStore(decimal.Zero, r.rateCard)
+	// always indicate that reserve was called.
+	r.reserved = true
```

Comment:
Part of why I mentioned making the name more explicit. I probably would have assume `reserved` = "successful reserve call", but it's actually "was the reserve call attempted"